aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -341,17 +341,19 @@ Tensor& _clamp_min_out_cpu(Tensor& result, const Tensor& self, Scalar min) {
   return result;
 }
 
-static inline void mvlgamma_check(const Tensor& self, int64_t p) {
-  TORCH_CHECK(at::isFloatingType(self.scalar_type()),
-              "mvlgamma is not implemented for ", self.scalar_type());
+static inline void mvlgamma_check(const Tensor& self, int64_t p, bool floating_ufunc=false) {
+  if(!floating_ufunc)
+    TORCH_CHECK(at::isFloatingType(self.scalar_type()),
+            "mvlgamma is not implemented for ", self.scalar_type());
```

Comment:
This check could have been removed, but `mvlgamma_` (inplace call) should not support type promotion.

Reply:
What if we don't make any changes to mvlgamma?

Reply:
I think, it will be nice to have changes to mvlgamma:

`torch.mvlgamma(torch.tensor((1,), dtype=torch.int8), p=2)` throws `RuntimeError: mvlgamma is not implemented for Char` error. 
`torch.mvlgamma(torch.tensor((1,), dtype=torch.float32), p=2)` returns `tensor([1.447])`. 

Based on the above comparison, I believe `mvlgamma` should have this promotion? I unfortunately don't have contextual knowledge about this!

Reply:
Let's leave it alone for now. We can always change it later.